core/src/worker/workflow/mod.rs

```diff
@@ -460,8 +464,16 @@ impl Workflows {
                 {
                     // If request_eager_execution was already false, that means lang explicitly
                     // told us it didn't want to eagerly execute for some reason. So, we only
-                    // ever turn *off* eager execution if a slot is not available.
-                    if attrs.request_eager_execution {
+                    // ever turn *off* eager execution if a slot is not available or the activity
+                    // is scheduled on a different task queue.
+                    if attrs.request_eager_execution
+                        && attrs
+                            .task_queue
+                            .as_ref()
+                            .map(|q| q.name == self.task_queue)
+                            .unwrap_or_default()
+                    {
+                        // if attrs.request_eager_execution {
```

Comment:
Leftover comment, and this logic is wrong. You aren't setting `request_eager_execution` to false if the task queue is mismatched

Reply:
Yeah, I just saw this, fixing